android/src/main/java/com/th3rdwave/safeareacontext/SafeAreaViewShadowNode.kt

```diff
@@ -65,23 +66,27 @@ class SafeAreaViewShadowNode : LayoutShadowNode() {
     left = PixelUtil.toPixelFromDIP(left)
     val edges = localData.edges
     val insets = localData.insets
-    val insetTop: Float = if (edges.contains(SafeAreaViewEdges.TOP)) insets.top else 0.0f
-    val insetRight: Float = if (edges.contains(SafeAreaViewEdges.RIGHT)) insets.right else 0.0f
-    val insetBottom: Float = if (edges.contains(SafeAreaViewEdges.BOTTOM)) insets.bottom else 0.0f
-    val insetLeft: Float = if (edges.contains(SafeAreaViewEdges.LEFT)) insets.left else 0.0f
+    val insetTop: Float = if (edges.top != SafeAreaViewEdgeModes.OFF ) insets.top else 0.0f
+    val insetRight: Float = if (edges.right != SafeAreaViewEdgeModes.OFF) insets.right else 0.0f
+    val insetBottom: Float = if (edges.bottom != SafeAreaViewEdgeModes.OFF) insets.bottom else 0.0f
+    val insetLeft: Float = if (edges.left != SafeAreaViewEdgeModes.OFF) insets.left else 0.0f
     if (localData.mode == SafeAreaViewMode.PADDING) {
-      super.setPadding(Spacing.TOP, insetTop + top)
-      super.setPadding(Spacing.RIGHT, insetRight + right)
-      super.setPadding(Spacing.BOTTOM, insetBottom + bottom)
-      super.setPadding(Spacing.LEFT, insetLeft + left)
+      super.setPadding(Spacing.TOP, getEdgeValue(edges.top, insetTop, top))
+      super.setPadding(Spacing.RIGHT, getEdgeValue(edges.right, insetRight, right))
+      super.setPadding(Spacing.BOTTOM, getEdgeValue(edges.bottom, insetBottom, bottom))
+      super.setPadding(Spacing.LEFT, getEdgeValue(edges.left, insetLeft, left))
     } else {
-      super.setMargin(Spacing.TOP, insetTop + top)
-      super.setMargin(Spacing.RIGHT, insetRight + right)
-      super.setMargin(Spacing.BOTTOM, insetBottom + bottom)
-      super.setMargin(Spacing.LEFT, insetLeft + left)
+      super.setMargin(Spacing.TOP, getEdgeValue(edges.top, insetTop, top))
+      super.setMargin(Spacing.RIGHT, getEdgeValue(edges.right, insetRight, right))
+      super.setMargin(Spacing.BOTTOM, getEdgeValue(edges.bottom, insetBottom, bottom))
+      super.setMargin(Spacing.LEFT, getEdgeValue(edges.left, insetLeft, left))
     }
   }
 
+  private fun getEdgeValue(edgeMode: SafeAreaViewEdgeModes, insetValue: Float, edgeValue: Float): Float {
+    return if (edgeMode == SafeAreaViewEdgeModes.MAXIMUM ) max(insetValue, edgeValue) else insetValue + edgeValue
```

Comment:
You could check for `edgeMode === SafeAreaViewEdgeModes.OFF` here too and remove lines 69-72